src/Sentry/Internal/Http/CachingTransport.cs

```diff
@@ -359,6 +383,26 @@ await using (stream.ConfigureAwait(false))
         _fileSystem.DeleteFile(file);
     }
 
+    /// <summary>
+    /// Send any files that were stuck in processing while the network was unavailable
+    /// </summary>
+    private void SendFilesQueuedWhileOffline()
+    {
+        if (Interlocked.Exchange(ref _networkIsUnavailable, 0) == 1)
+        {
+            // Stop any connectivity monitor that we might have started
+            lock (_connectivityMonitorLock)
+            {
+                _connectivityMonitor?.Dispose();
+                _connectivityMonitor = null;
+            }
+
+            // Retry envelopes that got stuck in processing while the transport was failing
+            MoveUnprocessedFilesBackToCache();
+            _workerSignal.Release();
+        }
+    }
```

Comment:
If we pause processing, then we don't need to move he files back to the cache on resume, right?

Reply:
Not sure about that... I think the first message that fails to send due to a network connectivity error will end up stuck in the `__processing` folder - but will test.